util/java/libtiled-java/src/main/java/org/mapeditor/io/TMXMapReader.java

```diff
@@ -219,10 +235,15 @@ private TileSet unmarshalTilesetFile(InputStream in, URL file) throws Exception
             tsNode = tsNodeList.item(0);
             if (tsNode != null) {
                 set = unmarshalTileset(tsNode);
-                if (set.getSource() != null) {
-                    System.out.println("Recursive external tilesets are not supported.");
+
+                // if using cached tilesets - we should already have correct source
+                if (!settings.reuseCachedTilesets) {
```

Comment:
If we don't ever assign the source when `reuseCachedTilesets` is true, then how would it already have a correct source?